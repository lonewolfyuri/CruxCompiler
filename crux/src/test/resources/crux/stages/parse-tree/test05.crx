// test expression

func main() : void {
  let a = 4 * 4;
  let b = 4 and 4;
  let c = 4 / 2 * 6 / 7 * true;
}
