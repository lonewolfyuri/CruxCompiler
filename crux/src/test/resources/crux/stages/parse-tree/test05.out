program
  declarationList
    declaration
      functionDefinition
        parameterList
        type
        statementBlock
          statementList
            statement
              assignmentStatement
                designator
                expression0
                  expression1
                    expression2
                      expression3
                        literal
                      op2
                      expression3
                        literal
            statement
              assignmentStatement
                designator
                expression0
                  expression1
                    expression2
                      expression3
                        literal
                      op2
                      expression3
                        literal
            statement
              assignmentStatement
                designator
                expression0
                  expression1
                    expression2
                      expression3
                        literal
                      op2
                      expression3
                        literal
                      op2
                      expression3
                        literal
                      op2
                      expression3
                        literal
                      op2
                      expression3
                        literal
